File: backend/Services/Parsers/PoeltlParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class PoeltlParser : BasicScoreResultParser
    {
        private ILogger<PoeltlParser> _logger;

        public PoeltlParser(ILogger<PoeltlParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Poeltl";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} \\d+ - (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => null;
        public override string Url => "https://poeltl.dunk.town/";
    }
}

File: lambdas/src/parsers/poeltl-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class PoeltlParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Poeltl';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://poeltl.dunk.town/';
  
  // Matches "Poeltl 700 - X/8" or "Poeltl 86 - 1/8"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} \\d+ - (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
